### PR TITLE
fix(perf-issues): Fix subject substitution for Performance issues

### LINKED ISSUE #46872: Subject substitution issue on email alert for Performance issue
The email I received 2 days ago showed the subject like:

image

Docs indicate the environment should be replaced: 

> You’re able to change the subject line for alert emails within your Project’s Email Settings by going to Project Settings > Alerts > Settings. Email subjects can include variables such as $project, $title, and ${tag:key}, such as ${tag:environment} or ${tag:release}.

I confirmed `environment` exists [in the event:](https://testorg-az.sentry.io/issues/4056375822/?referrer=alert_email&alert_type=email&alert_timestamp=1680447601889&alert_rule_id=889972) 

image

### PR DESCRIPTION
**Pull request by gggritso**
Closes #46872. Two changes here:

1. When someone puts in `$title`, for a Performance Issue it should use the issue type name (e.g., "Slow DB Query") since each event doesn't have its own title of any kind (except if there's an occurrence, but that's another story)
2. Use the aliased _and_ unaliased form of the tag. I'm not sure what the root cause is, but when users set `${tag:release}`, we look for the `sentry:release` tag, instead of the `release` tag. For Performance issues, this doesn't seem to work, that tag isn't there. I'm honestly not too sure what's going on, but I set it so that it also tries the unaliased tag before giving up

### CODE REVIEW

File: tests/sentry/eventstore/test_models.py [hunk 1]
```diff
@@ -105,6 +106,34 @@ def test_email_subject_with_template(self):
...
+    def test_transaction_email_subject(self):
+        min_ago = iso_format(before_now(minutes=1))
+        group = self.create_group(type=PerformanceSlowDBQueryGroupType.type_id)
+
+        self.project.update_option(
+            "mail:subject_template",
+            "$shortID - ${tag:environment}@${tag:release} $title",
+        )
+
+        mock_event = Event(
```
Comment (ceorourke) [inline tests/sentry/eventstore/test_models.py:118]: If you subclass the `PerformanceIssueTestCase` you can do `event = self.create_performance_issue()`

**Review by ceorourke — APPROVED** [review]
The test isn't blocking but it'd be good to use the util.  Thanks for fixing something I should have implemented 🙃

File: tests/sentry/eventstore/test_models.py (hunk 1 above)
Comment (gggritso) [reply tests/sentry/eventstore/test_models.py]: Wow that is a _lot_ better!

**Comment by gggritso** [comment]
@ceorourke 🙏🏻

**Pull request merged**